cherrypy/_cpreqbody.py

```diff
@@ -724,14 +738,22 @@ def default_proc(self):
                 self.file = result
 
     def read_into_file(self, fp_out=None):
-        """Read the request body into fp_out (or make_file() if None).
+        """Read the request body into fp_out.
+
+        if fp_out not given, use either make_file or
+        a BytesIO object, based on maxrambytes.
 
         Return fp_out.
         """
-        if fp_out is None:
-            fp_out = self.make_file()
-        self.read_lines_to_boundary(fp_out=fp_out)
-        return fp_out
+        if not fp_out:
+            x = self.read_lines_to_boundary()
+            if isinstance(x,(str,bytes)):
```

Comment:
`str` is alias of unicode type in Python 3 and bytes in Python 2. I'd go for try/except following [EAFP principle](https://docs.python.org/3/glossary.html).